tests/tests_integration/test_api/test_geospatial.py

```diff
@@ -449,6 +477,34 @@ def test_stream_features(self, cognite_client, large_feature_type, many_features
         feature_list = FeatureList(list(features))
         assert len(feature_list) == len(many_features)
 
+    def test_stream_features_dimensionality_mismatch(self, cognite_client, test_feature_type, test_feature):
+        polygon_z = "POLYGONZ((2.276 48.858 3,2.278 48.859 3,2.2759 48.859 3,2.276 48.858 3))"
+        polygon = "POLYGON((2.276 48.858,2.278 48.859,2.275 48.859,2.276 48.858))"
+        stream_res = cognite_client.geospatial.stream_features(
+            feature_type_external_id=test_feature_type.external_id,
+            filter={"stWithin": {"property": "position", "value": {"wkt": polygon}}},
+        )
+        res = [x for x in stream_res]
+        assert res[0].external_id == test_feature.external_id
+
+        try:
+            stream_res = cognite_client.geospatial.stream_features(
+                feature_type_external_id=test_feature_type.external_id,
+                filter={"stWithin": {"property": "position", "value": {"wkt": polygon_z}}},
+            )
+            _ = [x for x in stream_res]
+            raise pytest.fail("searching features with wrong dimensionality should raise exception")
```

Comment:
Is the list comprehension there to exhaust the generator? Add a small comment if so. Reove the raise keyword as well
```suggestion
            _ = [x for x in stream_res]
            pytest.fail("searching features with wrong dimensionality did not raise an exception")
```